File: java64/nivel4/exer37.java
import javax.swing.JOptionPane;

public class exer37 {
    private static double valor1, valor2;

    public static void main(String[] args) {
        int opcao2;
        String opcao1;
        boolean continua = false;

        while (continua == false) {
            opcao2 = Integer.parseInt(JOptionPane.showInputDialog(
                    "Escolha a opção: \n1 – Adição \n2 – Subtração \n3 – Multiplicação \n4 – Divisão "));

            switch (opcao2) {
                case 1:
                    valores();
                    JOptionPane.showMessageDialog(null, "Resultado da adição é: " + (valor1 + valor2));
                break;
                case 2:
                    valores();
                    JOptionPane.showMessageDialog(null, "Resultado da subtração é: " + (valor1 - valor2));
                break;
                case 3:
                    valores();
                    JOptionPane.showMessageDialog(null, "Resultado da multiplicação é: " + (valor1 * valor2));
                break;
                case 4:
                    valores();
                    JOptionPane.showMessageDialog(null, "Resultado da divisão é: " + (valor1 / valor2));
                break;
            }
            opcao1 = JOptionPane.showInputDialog("Deseja fazer outra operação? \nS ou s para sim\nN ou n para não");
            if (opcao1 == "N" || opcao1 == "n") continua = true;
        }
    }

    public static void valores() {
        valor1 = Integer.parseInt(JOptionPane.showInputDialog("Informe o primeiro valor: "));
        valor2 = Integer.parseInt(JOptionPane.showInputDialog("Informe o segundo valor: "));
    }
}

File: pyhton64/nivel4/exer37.py
def main():
    continua = False
    while continua == False:
        op = opcao1()
        if op == 1:
            valor1 = float(input('Valor 1: '))
            valor2 = float(input('Valor 1: '))
            adicao(valor1, valor2)

        elif op == 2:
            valor1 = float(input('Valor 1: '))
            valor2 = float(input('Valor 1: '))
            subtracao(valor1, valor2)

        elif op == 3:
            valor1 = float(input('Valor 1: '))
            valor2 = float(input('Valor 1: '))
            multiplicacao(valor1, valor2)

        elif op == 4:
            valor1 = float(input('Valor 1: '))
            valor2 = float(input('Valor 1: '))
            divisao(valor1, valor2)
        else: 
            print(('Opção invalida'))
        op2 = opcao2()
        if op2 == 'N' or op2 == 'n': 
            continua = True

def opcao1():
    opc = int(input('1 – Adição\n'+
        '2 – Subtração\n'+
        '3 – Multiplicação\n'+ 
        '4 – Divisão\n'+
        'Informe a opção: '))
    return opc

def opcao2():
    opc = str(input('\nS - s / N - n\nDeseja continuar? '))
    return opc

def adicao(valor1, valor2):
    soma = valor1 + valor2
    return print(f'Resultado: {soma}')

def subtracao(valor1, valor2):
    sub = valor1 - valor2
    return print(f'Resultado: {sub}')

def multiplicacao(valor1, valor2):
    mult = valor1 * valor2
    return print(f'Resultado: {mult}')

def divisao(valor1, valor2):
    div = valor1 / valor2
    return print(f'Resultado: {div}')

if __name__ == ("__main__"):
    main()